### DISCUSSION TITLE
Transform irradiance to power

### DISCUSSION THREAD

**Discussion by markkvdb** [open-meteo/open-meteo#648] [General]
The open-meteo forecast API provides the (great!) option to get the `global_tilted_irradiance` based on the panel configuration. The output is the power on the surface (W/m2). I'd like to find a way to transform this to the power output of the plant.

One first idea is to get the surface of the panels and data about the efficiency of the panel + losses, but this data is often unavailable. Instead, I often have access to the maximum power and the inverter loss. Does anyone have any ideas how I can come up with some kind of heuristic algorithm to do this transformation?

My first idea is to take the irradiance and observed power data for some reference days and use the ratio between the two to scale the irradiance data to power. However, this approach seems quite brittle.

**Comment by patrick-zippenfenig** [comment]
This could be start: https://github.com/openclimatefix/Open-Source-Quartz-Solar-Forecast. It is not yet using GTI or even DNI, but you could contact the authors over there.

In my experience every PV site is different and needs adjustments to get a good forecast. If you have access to measurements from an inverter you can use ML approaches or a simple multilinear regression to get a forecast.

**Reply by patrick-zippenfenig** [reply]
Good overview: https://fosdem.org/2024/schedule/event/fosdem-2024-2960-quartz-solar-os-building-an-open-source-ai-solar-forecast-for-everyone/

**Reply by markkvdb** [reply]
Thank you for the links. I'll check them out and show some insights if I find something useful that others might benefit from!

**Reply by markkvdb** [reply]
How do they even manage to produce good forecast without information about either the panel or historical data from the site you want to predict?

EDIT: they use some default values for the tilt and azimuth.

It seems that the training of the model is not available - just the binary. The organisation looks very interesting though and I will definitely point a few colleagues of mine to their work.

**Reply by patrick-zippenfenig** [reply]
There is an open PR to integrate panel PV data: https://github.com/openclimatefix/Open-Source-Quartz-Solar-Forecast/issues/36

Yes, I also noticed that cloud forecasts from GFS can be quite bad...
